### PR TITLE
Manage words_answers

### PR DESCRIPTION
**Pull request by tieppt07**

### CODE REVIEW

File: app/Http/Controllers/WordsController.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $categories = Category::lists('name', 'id')->all();
```
Comment (KhanhLD) [inline app/Http/Controllers/WordsController.php]: how about remove all() ?

File: app/Http/Controllers/WordsController.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $valid = Validator::make($request->all(), [
```
Comment (KhanhLD) [inline app/Http/Controllers/WordsController.php]: this part and the part in update are duplicated with each other.
how about move it to somewhere so both actions can use it? for example in model.

File: app/Http/Controllers/WordsController.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        ]);
+        if ($valid->fails()) {
+            return back()->withErrors($valid->errors());
+        } else {
+            $word = new Word();
+            $word->name = $request->get('name');
+            $word->description = $request->get('description');
+            $word->category_id = $request->get('category');
+            $word->save();
+            $data = $request->get('answer');
```
Comment (KhanhLD) [inline app/Http/Controllers/WordsController.php]: do not name variable $data or something hard to understand what it is.
name it $answers

File: app/Http/Controllers/WordsController.php [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $word = Word::findorFail($id);
+        $categories = Category::all();
```
Comment (KhanhLD) [inline app/Http/Controllers/WordsController.php]: use lists()

File: app/Http/Controllers/WordsController.php [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        } else {
+            $word = Word::findorFail($id);
+            $word->name = $request->get('name');
+            $word->description = $request->get('description');
+            $word->category_id = $request->get('category');
+            $word->save();
+            $data = $request->get('answer');
+            $answer = new Answer();
+            foreach ($data as $key) {
+                $answer->where('word_id', '=', $word->id)->update($key);
```
Comment (KhanhLD) [inline app/Http/Controllers/WordsController.php]: you can write where('word_id', $word->id) without '='

File: app/Http/Controllers/WordsController.php [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    /**
+     * Remove the specified resource from storage.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        $word = Word::findorFail($id);
+        $word->delete();
```
Comment (KhanhLD) [inline app/Http/Controllers/WordsController.php]: if a word is deleted, please also delete its answers

File: resources/views/admin/word/create.blade.php [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+@section('title', 'Create new Word')
+@section('page-header', 'Add New Word')
+@section('content')   
+    <div class="row-fluid">
+        <div class="well">
+            <div id="myTabContent" class="tab-content">
+                <div class="tab-pane active in" id="home">
+                    {!! Form::open(['url' => 'admin/words']) !!}
+                        @if (count($errors) > 0)
```
Comment (KhanhLD) [inline resources/views/admin/word/create.blade.php]: this part of showing errors is duplicated with edit view.
make it a partial to include in all needed views

File: resources/views/admin/word/create.blade.php [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+                        <div class="row">
+                            <div class="form-group col-xs-12 col-sm-12">
+                                {!! Form::label('name', 'Word:', ['class' => 'control-label']) !!}
+                                {!! Form::text('name', null, ['class' => 'form-control']) !!}
+                            </div>
+                            <div class="form-group col-xs-12 col-sm-12">
+                                {!! Form::label('category', 'Category:', ['class' => 'control-label']) !!}
+                                {!! Form::select('category', $categories, ['class' => 'form-control']) !!}
+                            </div>
+                            @for ($i = 0; $i < config('constant.answer_inputs'); $i++)
```
Comment (KhanhLD) [inline resources/views/admin/word/create.blade.php]: use range with foreach

File: resources/views/admin/word/edit.blade.php [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+                        {{$word->answers}}
+                        <div class="row">
+                            <div class="form-group col-xs-12 col-sm-12">
+                                {!! Form::label('name', 'Word:', ['class' => 'control-label']) !!}
+                                {!! Form::text('name', $word->name, ['class' => 'form-control']) !!}
+                            </div>
+                            <div class="form-group col-xs-12 col-sm-6">
+                                {!! Form::label('category', 'Category:', ['class' => 'control-label']) !!}
+                                <select name="category" id="">
+                                    @foreach ($categories as $category)
```
Comment (KhanhLD) [inline resources/views/admin/word/edit.blade.php]: please see here and update
http://laravelcollective.com/docs/5.1/html#drop-down-lists

File: app/Http/Controllers/AnswersController.php [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+     * Remove the specified resource from storage.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy(Request $request, $id)
+    {
+        $answer = Answer::findorFail($id);
+        $word_id = $request->get('word_id');
+        $answer->delete();
```
Comment (wataridori) [inline app/Http/Controllers/AnswersController.php]: ở đây em phải dùng 2 câu lệnh, 2 query DB
```php
$answer = Answer::findorFail($id);
$answer->delete();
```

sao không dùng luôn một câu `destroy` thôi em 
```php
Answer::destroy($id);
```

File: app/Http/Controllers/AnswersController.php [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+    /**
+     * Remove the specified resource from storage.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy(Request $request, $id)
+    {
+        $answer = Answer::findorFail($id);
+        $word_id = $request->get('word_id');
```
Comment (wataridori) [inline app/Http/Controllers/AnswersController.php]: Tên variable viết ở dạng `camelCase` nhé em

File: app/Http/Controllers/WordsController.php [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $word = Word::findorFail($id);
+        $answers = $word->answers()->get();
```
Comment (wataridori) [inline app/Http/Controllers/WordsController.php]: `$word->answers()->get();` có thể viết ngắn gọn là `$word->answers`

**Comment by wataridori** [comment]
OK

**Pull request merged**